examples/multi_cn/s0/run.sh

```diff
@@ -0,0 +1,237 @@
+average_num=30
+decode_modes="ctc_greedy_search ctc_prefix_beam_search attention attention_rescoring"
+
+. tools/parse_options.sh || exit 1;
+
+if [ ${stage} -le -1 ] && [ ${stop_stage} -ge -1 ]; then
+    echo "stage -1: Data Download"
+    # download all training data
+    local/aidatatang_download_and_untar.sh $dbase/aidatatang $aidatatang_url aidatatang_200zh || exit 1;
+    local/aishell_download_and_untar.sh $dbase/aishell $aishell_url data_aishell || exit 1;
+    local/magicdata_download_and_untar.sh $dbase/magicdata $magicdata_url train_set || exit 1;
+    local/primewords_download_and_untar.sh $dbase/primewords $primewords_url || exit 1;
+    local/stcmds_download_and_untar.sh $dbase/stcmds $stcmds_url || exit 1;
+    local/thchs_download_and_untar.sh $dbase/thchs $thchs_url data_thchs30 || exit 1;
+
+    # download all test data
+    local/thchs_download_and_untar.sh $dbase/thchs $thchs_url test-noise || exit 1;
+    local/magicdata_download_and_untar.sh $dbase/magicdata $magicdata_url dev_set || exit 1;
+    local/magicdata_download_and_untar.sh $dbase/magicdata $magicdata_url test_set || exit 1;
+fi
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    # Data preparation
+    local/aidatatang_data_prep.sh $dbase/aidatatang/aidatatang_200zh data/aidatatang || exit 1;
+    local/aishell_data_prep.sh $dbase/aishell/data_aishell data/aishell || exit 1;
+    local/thchs-30_data_prep.sh $dbase/thchs/data_thchs30 data/thchs || exit 1;
+    local/magicdata_data_prep.sh $dbase/magicdata/ data/magicdata || exit 1;
+    local/primewords_data_prep.sh $dbase/primewords data/primewords || exit 1;
+    local/stcmds_data_prep.sh $dbase/stcmds data/stcmds || exit 1;
+    if $has_aishell2; then
+        local/aishell2_data_prep.sh $dbase/aishell2/IOS data/aishell2/train || exit 1;
+        local/aishell2_data_prep.sh $dbase/aishell2/IOS/dev data/aishell2/dev || exit 1;
+        local/aishell2_data_prep.sh $dbase/aishell2/IOS/test data/aishell2/test || exit 1;
+    fi
+
+    tools/combine_data.sh data/train \
+        data/{aidatatang,aishell,magicdata,primewords,stcmds,thchs,aishell2}/train || exit 1;
+    tools/combine_data.sh data/dev \
+        data/{aidatatang,aishell,magicdata,thchs,aishell2}/dev || exit 1;
+fi
+
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    # For wav feature, just copy the data. Fbank extraction is done in training
+    mkdir -p $feat_dir
+
+    # remove the space between the text labels for Mandarin dataset
+    for x in train dev; do
+        cp data/${x}/text data/${x}/text.org
+        paste -d " " <(cut -f 1 -d" " data/${x}/text.org) <(cut -f 2- -d" " data/${x}/text.org | tr -d " ") \
+            > data/${x}/text
+        rm data/${x}/text.org
+        cp -r data/$x $feat_dir
+    done
+    
+    for x in ${test_sets}; do
+        cp data/${x}/test/text data/${x}/test/text.org
+        paste -d " " <(cut -f 1 -d" " data/${x}/test/text.org) <(cut -f 2- -d" " data/${x}/test/text.org | tr -d " ") \
```

Comment:
any special process of english words here?
I think it is required here, for example, "how are you" should be processed to "how▁are▁you", where ▁ is a special char for space between english words.

Reply:
Maybe you are right.
But in kaldi's recipe, they transfer all Chinese words and English words to CMU phoneme, But do not add special char for space in words of English, If we add special char, How about read two or more words without interval? like "Not at all".  and There are a lot of words like 'NBA' was read as 'N' 'B' 'A'. I am not sure what is useful for these cases in End2End training.

Reply:
Really great job.  E2E is different to kaldi where space doesn't require special process. If we don't special symbol for space, we can't know the english word boundary in inference since we doesn't train it at all.

Reply:
Maybe I can use spm_encode from librispeech for the English part?